addons/account/models/account_journal.py

```diff
@@ -92,8 +92,13 @@ def _default_invoice_reference_model(self):
     default_account_id = fields.Many2one(
         comodel_name='account.account', check_company=True, copy=False, ondelete='restrict',
         string='Default Account',
-        domain="[('deprecated', '=', False),"
-               "'|', ('account_type', '=', default_account_type), ('account_type', 'not in', ('asset_receivable', 'liability_payable'))]")
+        domain="""[
+            ('deprecated', '=', False),
+            '|', ('account_type', '=', default_account_type),
```

Comment:
Now that I'm reading this, what does that mean? :thinking: 
`'|', ('account_type', '=', default_account_type),`

I see that `default_account_type` can only be from `asset_cash`/`income`/`expense`, so never `asset_receivable`/`liability_payable` anyway... So that OR is useless?

Reply:
There is indeed an issue here that was brought from this commit: https://github.com/odoo/odoo/commit/3312657294947b1cc8670a0ed3557e12ee6969c8. For example when using other journal types, you can select any account we want, so the `default_account_type` field seems to be useless. I discussed it with Benj and for now he only wants to solve the receivable/payable problem only for the misc journals.